sos/plugins/sapnw.py

```diff
@@ -0,0 +1,139 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+import os
+from sets import Set
+from sos.plugins import Plugin, RedHatPlugin
+
+
+def get_directory_listing(path):
+    try:
+        dir_list = os.listdir(path)
+    except:
+        dir_list = []
+    return dir_list
+
+
+class sapnw(Plugin, RedHatPlugin):
+    """SAP NetWeaver"""
+
+    plugin_name = 'sapnw'
+    profiles = ['sap']
+
+    files = ['/usr/sap']
+
+    def setup(self):
+
+        # list installed instances
+        inst_out = self.get_cmd_output_now("/usr/sap/hostctrl/exe/saphostctrl \
+                                           -function ListInstances",
+                                           suggest_filename="SAPInstances")
+        # list installed sap dbs
+        db_out = self.get_cmd_output_now("/usr/sap/hostctrl/exe/saphostctrl \
+                                         -function ListDatabases",
+                                         suggest_filename="SAPDatabases")
+
+        sidsunique = Set([])
+
+        # Cycle through all the instances, get 'sid' 'instance_number'
+        # and 'vhost' to determine the proper profile
+        p = open(inst_out, "r").read().splitlines()
+        for line in p:
+            if "DAA" not in line:
+                fields = line.strip().split()
+                sid = fields[3]
+                inst = fields[5]
+                vhost = fields[7]
+                sidsunique.add(sid)
+                for line in get_directory_listing("/usr/sap/%s/SYS/profile/"
+                                                  % sid):
```

Comment:
I think the 'pass' was just inadvertently copied from my example code (an empty for body is syntactically invalid so rather than copy in the whole of the original I just substitude a `pass`).